subgraphs/biswap/src/common/creators.ts

```diff
@@ -0,0 +1,405 @@
+import { Address, BigDecimal, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import {
+  Deposit,
+  LiquidityPool,
+  LiquidityPoolFee,
+  Swap,
+  Withdraw,
+  _HelperStore,
+  _LiquidityPoolAmount,
+} from "../../generated/schema";
+import {
+  BIGDECIMAL_FIFTY,
+  BIGDECIMAL_NEG_ONE,
+  BIGDECIMAL_TWO,
+  BIGDECIMAL_ZERO,
+  BIGINT_NEG_ONE,
+  BIGINT_ZERO,
+  BISWAP_LP_FEE,
+  BISWAP_PROTOCOL_FEE,
+  BISWAP_TRADING_FEE,
+  INT_ONE,
+  INT_ZERO,
+  LiquidityPoolFeeType,
+} from "./constants";
+import {
+  getLiquidityPool,
+  getLiquidityPoolAmounts,
+  getOrCreateDex,
+  getOrCreateLPToken,
+  getOrCreateToken,
+} from "./getters";
+import { updateVolumeAndFees } from "./updateMetrics";
+import { convertFeeToPercent, convertTokenToDecimal } from "./utils";
+
+export function createLiquidityPool(
+  event: ethereum.Event,
+  poolAddress: string,
+  token0Address: string,
+  token1Address: string
+): void {
+  let protocol = getOrCreateDex();
+
+  // create the tokens and tokentracker
+  let token0 = getOrCreateToken(event, token0Address);
+  let token1 = getOrCreateToken(event, token1Address);
+  let poolToken = getOrCreateLPToken(poolAddress, token0, token1);
```

Comment:
Are we cool with grabbing the the name, symbol, decimals from the LP token contract directly? (see getOrCreateToken vs getOrCreateLPToken). I think that is better.

Reply:
Do we want to track the price of the poolToken/outputToken (using prices lib)? I believe the prices lib won't have price for most LP tokens and/or they may be inaccurate.